pages/bus_routes.jsx

```diff
@@ -0,0 +1,284 @@
+import React from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import {
+  Button,
+  ButtonGroup,
+  Fab,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  TextField
+} from "@material-ui/core";
+import EditIcon from "@material-ui/icons/Edit";
+import AddIcon from "@material-ui/icons/Add";
+import CheckCircleIcon from "@material-ui/icons/CheckCircle";
+
+const useStyles = makeStyles(() => ({
+  container: {
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "center",
+    alignItems: "center"
+  },
+  routeNameContainer: {
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "center",
+    justifyContent: "center"
+  },
+  routeName: {
+    fontSize: "30px",
+    fontWeight: "bold",
+    color: "black"
+  },
+  icon: {
+    color: "#B3B8BA",
+    padding: "0px 0px 0px 25px"
+  },
+  checkIcon: {
+    color: "#3fd95b",
+    padding: "0px 0px 0px 25px"
+  },
+  hideIcon: {
+    display: "none"
+  },
+  btn: {
+    textDecoration: "none",
+    textAlign: "center",
+    color: "black",
+    margin: "10px",
+    padding: "10px",
+    background: "#BDBDBD"
+  },
+  pagehead: {
+    width: "95%",
+    display: "flex",
+    flexDirection: "row",
+    justifyContent: "space-between",
+    alignItems: "center"
+  },
+  table: {
+    width: "100%",
+    borderSpacing: "5px",
+    textAlign: "center",
+    borderCollapse: "collapse"
+  },
+  tbody: {
+    display: "block",
+    height: "450px",
+    overflowY: "auto",
+    overflowX: "hidden"
+  },
+  th: {
+    width: "calc( 100% - 1em )",
+    backgroundColor: "#E0E0E0",
+    padding: "10px",
+    border: "1px solid #BDBDBD",
+    borderCollapse: "collapse"
+  },
+  td: {
+    textAlign: "center",
+    width: "fill",
+    padding: "5px",
+    borderLeft: "1px solid #BDBDBD",
+    borderRight: "1px solid #BDBDBD"
+  },
+  tr: {
+    display: "table",
+    width: "100%",
+    tableLayout: "fixed",
+    "&:nth-child(even)": {
+      backgroundColor: "#efefef"
+    }
+  },
+  input: {
+    margin: "25px",
+    padding: "10px"
+  },
+  routeTabs: {
+    display: "flex"
+  },
+  label: {
+    fontSize: "16px",
+    fontWeight: "bold"
+  },
+  textField: {
+    border: "none",
+    backgroundColor: "#efefef",
+    display: "inline-block",
+    minHeight: "24px",
+    padding: "2px 8px",
+    margin: "10px",
+    fontSize: "14px"
+  }
+}));
+
+const BusRoutes = () => {
+  const classes = useStyles();
+  const [routes, setRoutes] = React.useState([
+    {
+      name: "route 1"
+    },
+    {
+      name: "route 2"
+    }
+  ]);
+  const [selectedRoute, setSelectedRoute] = React.useState(routes[0]);
+  const [tempRoute, setTempRoute] = React.useState(routes[0].name);
+  const [routeEditable, setEditable] = React.useState(false);
+  const [modalOpen, setModalOpen] = React.useState(false);
+
+  const addRoute = () => {
+    setModalOpen(true);
+  };
+
+  const handleCreate = () => {
+    setRoutes(routes.concat({ name: "New Route" }));
```

Comment:
Shouldn't the name be set to the bus route name entered in the modal?

Reply:
Just fixed it!